source/nsdynmemLIB/nsdynmemLIB.c

```diff
@@ -360,39 +403,45 @@ static void ns_free_and_merge_with_adjacent_blocks(int * const cur_block, int da
 }
 #endif
 
-void ns_dyn_mem_free(void *block)
+void ns_mem_free(ns_mem_book_t *book, void *block)
 {
 #ifndef STANDARD_MALLOC
-    int *ptr = block;
-    int size;
+    if(!book) {
```

Comment:
Oh, one little request - take this null test out. Let it crash if they pass null book.

(null block is legal, and supposed to be a no-op, but book must be valid).